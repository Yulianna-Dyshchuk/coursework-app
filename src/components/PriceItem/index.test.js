import React from "react";
import { shallow } from "enzyme";
import { PriceItem } from "./index";

describe("PriceItem component", () => {
  it("should render correctly without props", () => {
    const component = shallow(<PriceItem />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const component = shallow(
      <PriceItem
        title="TEST TITLE"
        description="TEST DESCRIPTION"
        price={1000}
        imageUrl="https://blog.pokupon.ua/wp-content/uploads/2019/08/image3-9.jpg"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
