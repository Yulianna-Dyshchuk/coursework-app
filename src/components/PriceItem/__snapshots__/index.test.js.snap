// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceItem component should render correctly with props 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundImage": "url(https://blog.pokupon.ua/wp-content/uploads/2019/08/image3-9.jpg)",
      }
    }
  >
    <div>
      <WithStyles(ForwardRef(Typography))
        variant="body2"
      >
        <b>
          1000
        </b>
         uah
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      TEST TITLE
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      TEST DESCRIPTION
    </WithStyles(ForwardRef(Typography))>
  </div>
</div>
`;

exports[`PriceItem component should render correctly without props 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div>
      <WithStyles(ForwardRef(Typography))
        variant="body2"
      >
        <b />
         uah
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    />
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    />
  </div>
</div>
`;
